worksheet_D/oxo.py

```diff
@@ -0,0 +1,176 @@
+BOARD_EDGE_LENGTH = 3
+WINNING_ROW_LENGTH = 3
+
+# Changing the board size obviously requires redefining the --+---+-- drawing section in show().
+# The winning row may not have to be the BOARD_EDGE_LENGTH so this is built in to the checks.
+#
+# oxotest.py does not allow lengths greater than 3 currently. I may tweak it to prove it works.
+#
+# Also would have written as BOARD_HEIGHT and BOARD_WIDTH but the worksheet asks for n * n implying square.
+
+
+class OxoBoard:
+    def __init__(self):
+        """ The initialiser. Initialise any fields you need here. """
+        self.board = {}
+        for x in xrange(BOARD_EDGE_LENGTH):
+            for y in xrange(BOARD_EDGE_LENGTH):
+                self.board.update({(x, y) : 0})
+
+    def get_square(self, x, y):
+        """ Return 0, 1 or 2 depending on the contents of the specified square. """
+        return self.board.get((x, y))
```

Comment:
Worth commenting why you use `get` instead of `[]` -- I think because elsewhere you rely on `get_square` returning `None` rather than throwing an exception for out-of-bounds coordinates?